### PR TITLE
Restructure front-end client to be standalone from flask

### PR DESCRIPTION
**Pull request by sohkai**
Restructures the front-end so that it's standalone and not dependent on the flask server to serve its files. 

Benefits:
- Complete separation between the front-end and back-end:
  - Front-end is completely standalone and can serve its own assets
  - Flask server is solely responsible for providing API views for the front-end
- Arguably easier to host somewhere since CORS has been dealt with
- It's clearer what each part is doing (and, honestly, how simple it is to talk to BigchainDB server-side)
- Front-end is hot reloadable, although this is somewhat hit or miss with the way we've set this up (ie. multiple entry points)
- Easier to build new apps on top since a few more things are automated through the webpack config.

May be interesting:
- The usage of environment variables to let `docker-compose` configure the front-end and back-end containers so that 1. the front-end is built with knowledge of where the back-end container is running, and 2. the back-end is built to accept the front-end origin for CORS. This is only really important for users who are running Docker through `docker-machine`, as they'll be using non-localhost ip addresses to access the containers, but it's a nice little trick.

### CODE REVIEW

File: client/app_index_template.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+<html lang="en">
+    <head>
+        <meta charset="UTF-8">
+        <title><%= htmlWebpackPlugin.options.title || 'BigchainDB Examples' %></title>
+        <meta name="viewport" content="initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no" />
+    </head>
+    <body>
+        <div id="bigchaindb-example-app"></div>
+    </body>
+</html
```
Comment (diminator) [inline client/app_index_template.html]: `</html>`

Comment (sohkai) [inline client/app_index_template.html]: Good call!

File: .env_template [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# App settings
+CLIENT_HOST=localhost
```
Comment (diminator) [inline .env_template:2]: `FLASK_BASE_PATH is not defined`

**Comment by diminator** [comment]
can you also update the README?
- go into `client` dir and do `npm install && npm start`

File: .env_template (hunk 2 above)
Comment (sohkai) [inline .env_template:2]: Good catch, although `FLASK_BASE_PATH` was replaced by `FLASK_BASE_URL` at some point down the line. I turns out I forgot to change it in the constants file though.

**Comment by sohkai** [comment]
Ahh, I assume the README instructions are going to get removed by #9 so I won't bother updating it. If these two look good otherwise, I'll merge both.

**Comment by diminator** [comment]
works! good to merge

**Pull request merged**